src/clients/tls_client.h

```diff
@@ -21,19 +21,85 @@
 #include <string>
 #include <vector>
 
+struct auto_free_mbedtls_net_context
```

Comment:
These structs should be `CamelCase` for consistency with the rest of our code, ie `AutoFreeMbedtlsNetContext`.